tests/virt/livemerge_test.py

```diff
@@ -878,6 +878,35 @@ def test_internal_merge():
     assert vm.volume_monitor.enabled
 
 
+def test_extend_base_bitmaps(fake_time):
+    config = Config('internal-merge')
+    sd_id = config.values["drive"]["domainID"]
+    img_id = config.values["drive"]["imageID"]
+    merge_params = config.values["merge_params"]
+    job_id = merge_params["jobUUID"]
+    top_id = merge_params["topVolUUID"]
+    base_id = merge_params["baseVolUUID"]
+
+    vm = RunningVM(config)
+
+    # Simulate required with top bitmaps does not require extend.
+    base = vm.cif.irs.prepared_volumes[(sd_id, img_id, base_id)]
+    vm.cif.irs.measure_info[(top_id, base_id)] = {
+        "required": base["apparentsize"],
+        "bitmaps": 0,
+    }
+    # Add base bitmaps not tracked in top measure.
+    # Merge should account for these bitmaps and extend.
+    vm.cif.irs.measure_info[(base_id, None)] = {
+        "bitmaps": 5 * MiB,
+    }
+    vm.merge(**merge_params)
+
+    # Merge persists job in EXTEND state.
+    persisted_job = parse_jobs(vm)[job_id]
+    assert persisted_job["state"] == Job.EXTEND
```

Comment:
It will be better to separate the test from the actual code so we can verify that the 
test really fail without the code change. Otherwise can have a broken test that does nothing.